drivers/overwrite_queue.go

```diff
@@ -0,0 +1,106 @@
+package drivers
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/common"
+)
+
+const (
+	timeoutMultiplier    = 1.5
+	overwriteQueueLength = 32
+)
+
+type (
+	OverwriteQueue struct {
+		desc           string
+		name           string
+		session        OSSession
+		maxRetries     int
+		initialTimeout time.Duration
+		maxTimeout     time.Duration
+		queue          chan []byte
+		quit           chan struct{}
+	}
+)
+
+func NewOverwriteQueue(session OSSession, name, desc string, maxRetries int, initialTimeout, maxTimeout time.Duration) *OverwriteQueue {
+	oq := &OverwriteQueue{
+		desc:           desc,
+		name:           name,
+		maxRetries:     maxRetries,
+		session:        session,
+		initialTimeout: initialTimeout,
+		maxTimeout:     maxTimeout,
+		queue:          make(chan []byte, overwriteQueueLength),
+		quit:           make(chan struct{}),
+	}
+	if maxRetries < 1 {
+		panic("maxRetries should be greater than zero")
+	}
+	go oq.workerLoop()
+	return oq
+}
+
+// Save queues data to be saved
+func (oq *OverwriteQueue) Save(data []byte) {
+	oq.queue <- data
+}
+
+// StopAfter stops reading loop after some time
+func (oq *OverwriteQueue) StopAfter(pause time.Duration) {
+	glog.Infof("=======> stop after %s", pause)
```

Comment:
If we're including this I think it should be only for the VERBOSE verbosity level.

Reply:
removed - this one was for debugging